src/components/MultiHopTrade/hooks/useReceiveAddress.tsx

```diff
@@ -38,9 +39,10 @@ export const useReceiveAddress = ({
 
   // Selectors
   const buyAsset = useAppSelector(selectInputBuyAsset)
-  // No need to pass a sellAssetAccountId to synchronize the buy account here - by the time this is called, we already have a valid buyAccountId
+  const sellAssetAccountId = useAppSelector(selectFirstHopSellAccountId)
+
   const buyAccountId = useAppSelector(state =>
-    selectLastHopBuyAccountId(state, { accountId: undefined }),
+    selectLastHopBuyAccountId(state, { accountId: sellAssetAccountId }),
```

Comment:
If we don't pass this in we don't look for a matching account, and instead just use the account with the highest balance.

For this to match the logic in `useAccountIds`, namely:

https://github.com/shapeshift/web/blob/54426c45ac66936fcb3be6c091f6faa26542c3e3/src/components/MultiHopTrade/hooks/useAccountIds.tsx#L41-L47

we need to pass in an account id to try match against.